JVM resource jar creation is broken on Mac by variation in `touch` command.

BSD touch uses `-t` to set the timestamp whereas GNU touch uses `-d`. We use / assume the GNU binary as noted here: https://github.com/pantsbuild/pants/pull/16950#discussion_r1157196330

As discovered here: https://pantsbuild.slack.com/archives/C046T6T9U/p1680604327733559?thread_ts=1680604327.733559&cid=C046T6T9U

Where the error looks like:
```
 12:24:56.74 [ERROR] 1 Exception encountered:
 
 Engine traceback:
   in select
   in pants.core.goals.check.check
   in pants.backend.scala.goals.check.scalac_check (scalac)
   in pants.backend.scala.compile.scalac.compile_scala_source
   in pants.jvm.compile.compile_classpath_entries
   in pants.jvm.resources.assemble_resources_jar
   in pants.engine.process.fallible_to_exec_result_or_raise
 Traceback (most recent call last):
   File "/Users/jbenito/.cache/pants/setup/bootstrap-Darwin-x86_64/pants.1Nnv7r/install/lib/python3.9/site-packages/pants/engine/process.py", line 275, in fallible_to_exec_result_or_raise
     raise ProcessExecutionFailure(
 pants.engine.process.ProcessExecutionFailure: Process 'Build resources JAR for sdk/transport-security-web-lib/src/test/resources:resources' failed with exit code 1.
 stdout:
 
 stderr:
 /usr/bin/touch: illegal option -- d
 usage:
 touch [-A [-][[hh]mm]SS] [-acfhm] [-r file] [-t [[CC]YY]MMDDhhmm[.SS]] file ...
```

It appears #16950 was cherry picked back to 2.13.1 and 2.14.0; so Pants has been broken for JVM resource jars since 2.13.1.

--- a/src/python/pants/jvm/resources.py
+++ b/src/python/pants/jvm/resources.py
@@ -98,8 +98,9 @@ async def assemble_resources_jar(
                 "-c",
                 " ".join(
                     [
+                        "TZ=UTC",
                         touch.path,
-                        "-d 1980-01-01T00:00:00Z",
+                        "-t 198001010000.00",
                         input_filenames,
                         "&&",
                         "TZ=UTC",

